File: apps/server-car-rental/src/Infrastructure/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CarRentalService.Infrastructure.Models;

[Table("Customers")]
public class Customer
{
    [Key, Required]
    public long Id { get; set; }

    [Required]
    public DateTime CreatedAt { get; set; }

    [Required]
    public DateTime UpdatedAt { get; set; }

    public ICollection<Rental> Rentals { get; set; } = new List<Rental>();

    public string? Email { get; set; }

    public string? LicenseNumber { get; set; }

    public string? LastName { get; set; }

    public string? Phone { get; set; }

    public string? Address { get; set; }

    public string? FirstName { get; set; }
}

File: apps/admin-car-rental/src/api/customer/Customer.ts
import { Rental } from "../rental/Rental";
import { Order } from "../order/Order";
import { FeedbackRating } from "../feedbackRating/FeedbackRating";

export type Customer = {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  rentals?: Array<Rental>;
  email: string | null;
  licenseNumber: string | null;
  lastName: string | null;
  phone: string | null;
  address: string | null;
  firstName: string | null;
  orders?: Array<Order>;
  feedbackRatings?: Array<FeedbackRating>;
};
